### PR TITLE
Added trick or treating

### PR DESCRIPTION
**Pull request by emilyploszaj**
* Saying "trick or treat" around villages while wearing a costume will make them toss you some pie (eventually candy)
* Added a blaze skirt costume

### CODE REVIEW

**Comment by Shrimpstress** [comment]
Please merge master into your fork and fix merge conflicts.

**Comment by emilyploszaj** [comment]
Should be fixed

File: build.gradle [hunk 1]
```diff
@@ -2,13 +2,66 @@ plugins {
...
+	maven {
+		name = "JitPack"
+		url = "https://jitpack.io"
+	}
+	maven {
+		name = "NerdHubMC"
+		url = "https://maven.abusedmaster.xyz/"
+	}
+}
+
```
Comment (Shrimpstress) [inline build.gradle]: This already exists lower down in the build.gradle.

Edit: To clarify, repositories already exists in the build.gradle. I realize my wording was a bit weird here.

File: build.gradle [hunk 2]
```diff
@@ -2,13 +2,66 @@ plugins {
...
+	modApi ("io.github.prospector:modmenu:1.7.13+build.123") {
+		transitive = false
+	}
+	
+	modCompile "com.github.NerdHubMC.Cardinal-Components-API:cardinal-components-base:${cardinal_components_version}"
+	include "com.github.NerdHubMC.Cardinal-Components-API:cardinal-components-base:${cardinal_components_version}"
+	modCompile "com.github.NerdHubMC.Cardinal-Components-API:cardinal-components-entity:${cardinal_components_version}"
+	include "com.github.NerdHubMC.Cardinal-Components-API:cardinal-components-entity:${cardinal_components_version}"
+	modImplementation "com.github.emilyploszaj:trinkets:${trinkets_version}"
+}
```
Comment (Shrimpstress) [inline build.gradle]: This already exists lower down in the build.gradle.

Edit: To clarify, dependencies already exists in the build.gradle. I realize my wording was a bit weird here.

**Review by Shrimpstress — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (emilyploszaj) [reply build.gradle]: Reorganized things to not have duplicates

File: build.gradle (hunk 2 above)
Comment (emilyploszaj) [reply build.gradle]: Reorganized things to not have duplicates

**Review by Shrimpstress — APPROVED** [review]
Things look great to me! Something good for a future PR may be to expose `costumeItems` in `ServerPlayNetworkHandlerMixin` as a proper Registry eventually.

File: src/main/java/com/fabriccommunity/spookytime/common/CandyComponent.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import net.minecraft.entity.Entity;
+import net.minecraft.nbt.CompoundTag;
+import net.minecraft.nbt.ListTag;
+
+public interface CandyComponent extends Component {
+	public boolean canGiveCandy(Entity entity);
+
+	public void setLastCandyTime(Entity entity, long time);
+
+	public class VillagerCandyComponnt implements CandyComponent {
```
Comment (Shnupbups) [inline src/main/java/com/fabriccommunity/spookytime/common/CandyComponent.java]: Component is misspelled here

Comment (emilyploszaj) [reply src/main/java/com/fabriccommunity/spookytime/common/CandyComponent.java]: My broken E key finally got me, fixed

**Comment by Shnupbups** [comment]
Looks good! I agree that there should be a proper registry for both Costume Items and Candy Items, or perhaps candy items could be a loot table, but that can all be done in a later PR.

**Review by Shnupbups — APPROVED** [review]

**Review by ImAK9 — APPROVED** [review]
Looks fine! I’ll let you merge this one shnupuu ;)

**Pull request merged**